enterprise/backup/src/test/java/org/neo4j/commandline/admin/BackupUsageTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.commandline.admin;
+
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.Console;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.neo4j.backup.impl.BackupHelpOutput;
+import org.neo4j.backup.impl.OnlineBackupCommandProvider;
+import org.neo4j.backup.impl.ParametrisedOutsideWorld;
+import org.neo4j.io.fs.DefaultFileSystemAbstraction;
+import org.neo4j.io.fs.FileSystemAbstraction;
+
+import static org.junit.Assert.assertEquals;
+
+public class BackupUsageTest
+{
+    OnlineBackupCommandProvider onlineBackupCommandProvider = new OnlineBackupCommandProvider();
+    CommandLocator commandLocator = AugmentedCommandLocator.fromFixedArray( onlineBackupCommandProvider );
+
+    @Test
+    public void outputMatchesExpectedForMissingBackupDir() throws UnsupportedEncodingException
+    {
+        // when
+        String output = runBackup();
+
+        // then
+        String reason = "Missing argument 'backup-dir'\n\n";
+        assertEquals( reason + BackupHelpOutput.BACKUP_OUTPUT, output );
```

Comment:
These are going to be very brittle. Maybe use Matchers.containsString to sanity check the output, without breaking on every whitespace change.

Reply:
I agree with @andrewkerr9000. There is now need to be this strict. It's a responsibility of the command line framework to print help correctly. Here we are just interested in presence of "Missing argument..." string. Class `BackupHelpOutput` is also not very pretty and imho should be removed.

Reply:
I have specifically used test like this when I was refactoring in the past. The best part about the test was it was brittle -> led me to fix things and move on. W/o this test the only way I can figure out changes to the help text after a refactoring was by building neo4j. Please dont change it to contains check.